ui/packages/catalog-ui-search/src/main/webapp/js/model/User.js

```diff
@@ -437,8 +437,29 @@ User.Response = Backbone.AssociatedModel.extend({
   canRead(metacard) {
     return new Security(Restrictions.from(metacard)).canRead(this)
   },
-  canWrite(metacard) {
-    return new Security(Restrictions.from(metacard)).canWrite(this)
+  /**
+   * Allow workspaces to be created no matter what,
+   * but still enforce the other restrictions (if it's a shared workspace,
+   * check if they have permission to write to it)
+   *
+   * For all other types, every policy is enforced.
+   */
+  canWrite(thing) {
+    if (thing.type !== 'workspace' && properties.isEditingRestricted()) {
+      return false
+    }
+    switch (thing.type) {
+      case 'workspace':
+      case 'metacard-properties':
+        return new Security(Restrictions.from(thing)).canWrite(this)
+      case 'query-result':
+        return this.canWrite(thing.get('metacard').get('properties'))
+      case 'query-result.collection':
+      default:
+        return !thing.some(subthing => {
```

Comment:
I think this can be rewritten as 
```
thing.every(subthing => {
return this.canWrite(subthing)
})
```

It would also be nice to rename `thing` to a more specific variable name if possible.